cilib/enums.py

```diff
@@ -41,6 +41,25 @@ class Series(Enum):
     def __hash__(self):
         return hash(self.name)
 
+    @classmethod
+    def from_base(cls, base: str) -> "Series":
+        """Get the Series enum from a base string like 'ubuntu@22.04'."""
+        try:
+            distro, version = base.split("@", 1)
+        except ValueError:
+            raise ValueError(
+                f"Invalid base format: '{base}' (expected 'distro@version')"
+            )
+
+        if distro != "ubuntu":
+            raise ValueError(f"Unsupported base '{base}' for Series lookup")
+
+        for member in cls:
+            if version.endswith(member.value):
+                return member
```

Comment:
Is it possible to do something like this?
```suggestion
        version_tuple = tuple(map(int, version.split(".")))
        for member in cls:
            if member._version_tuple() == version_tuple:
                return member
```